libs/ha/ha_common_libs_k8s.py

```diff
@@ -1312,9 +1318,27 @@ def simulate_disk_cvg_failure(self, node_obj, source, resource_status, resource_
                 config_dict["events"][f"{count}"]["resource_type"] = resource_type
                 config_dict["events"][f"{count}"]["source"] = source
                 config_dict["events"][f"{count}"]["resource_status"] = resource_status
-                config_dict["events"][f"{count}"]["node_id"] = node_id_list[n_cnt]
-                config_dict["events"][f"{count}"]["resource_id"] = resource_id_list[d_cnt]
-
+                if node_id and resource_id:
+                    # If user provide node_id and resource_id
+                    config_dict["events"][f"{count}"]["node_id"] = node_id
+                    config_dict["events"][f"{count}"]["resource_id"] = resource_id
+                elif node_id and resource_type == 'node':
+                    # If user provide node_id
+                    config_dict["events"][f"{count}"]["node_id"] = node_id
+                    config_dict["events"][f"{count}"]["resource_id"] = node_id
+                elif resource_id and resource_type == 'node':
+                    # If user provide resource_id
+                    config_dict["events"][f"{count}"]["node_id"] = resource_id
+                    config_dict["events"][f"{count}"]["resource_id"] = resource_id
```

Comment:
Why do we need this validation? Validation should be already handled in the mock monitor script. If not then it will be a bug.

Reply:
This validation is needed to check the node_id and resource_id when user provide input values also it will take default value when user does not provide any values. This is will get update in config.json file. This has nothing to do with mock monitor script.